spark/src/main/kotlin/com/adevinta/spark/components/toggles/CheckBox.kt

```diff
@@ -0,0 +1,192 @@
+package com.adevinta.spark.components.toggles
+
+import androidx.compose.foundation.interaction.Interaction
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.state.ToggleableState
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import com.adevinta.spark.InternalSparkApi
+import com.adevinta.spark.PreviewTheme
+import com.adevinta.spark.tools.modifiers.minimumTouchTargetSize
+import com.adevinta.spark.tools.modifiers.sparkUsageOverlay
+import com.adevinta.spark.tools.preview.SparkPreviewParam
+import com.adevinta.spark.tools.preview.SparkPreviewParamProvider
+
+@Composable
+@InternalSparkApi
+internal fun SparkCheckbox(
+    state: ToggleableState,
+    onClick: (() -> Unit)?,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
+    androidx.compose.material3.TriStateCheckbox(
+        state = state,
+        onClick = onClick,
+        interactionSource = interactionSource,
+        enabled = enabled,
+        modifier = modifier.sparkUsageOverlay(),
+    )
+}
+
+/**
+ *
+ * Checkboxes allows users to select one or more items from a set. Checkboxes can turn an option on or off.
+ *
+ *  - Toggle a single item on or off.
+ *  - Require another action to activate or deactivate something.
+ *
+ * @see [SparkCheckbox] if you require support for an indeterminate state, or more advanced
+ * color customization between states. Be aware that this is still an internal composable so if you need such
+ * state contact the brikke team
+ *
+ * @param state whether TriStateCheckbox is checked, unchecked or in indeterminate state
+ * @param onCheckedChange callback to be invoked when checkbox is being clicked,
+ * therefore the change of checked state in requested.  If null, then this is passive
+ * and relies entirely on a higher-level component to control the "checked" state.
+ * @param modifier Modifier to be applied to the layout of the checkbox
+ * @param enabled whether the component is enabled or grayed out
+ * @param interactionSource the [MutableInteractionSource] representing the stream of
+ * [Interaction]s for this Checkbox. You can create and pass in your own remembered
+ * [MutableInteractionSource] if you want to observe [Interaction]s and customize the
+ * appearance / behavior of this Checkbox in different [Interaction]s.
+ */
+@Composable
+public fun Checkbox(
+    state: ToggleableState,
+    onCheckedChange: (() -> Unit)?,
```

Comment:
We "can't" use onCheckedChange with boolean value since `ToggleableState` so I think we should use `onClick` but it'll break the existing api when named arguments are used 😞 

Migrating to the new naming should still be feasible if we do it soon 🤔